scripts/tasks/version-check.js

```diff
@@ -6,9 +6,63 @@
  */
 
 'use strict';
+const execFileSync = require('child_process').execFileSync;
+const yarnCmds = ['yarnpkg', 'yarn'];
+
+function getVersion(cmd) {
+  return execFileSync(cmd, ['--version'], {
+    encoding: 'utf-8',
+  }).trim();
+}
+
+function getLocalYarnVersion() {
+  let yarnVersion;
+  let caughtError = null;
+
+  const isYarnInstalled = yarnCmds.some(function(cmd) {
+    try {
+      yarnVersion = getVersion(cmd);
+      return true;
+    } catch (err) {
+      caughtError = err;
+      return false;
+    }
+  });
+
+  if (isYarnInstalled) {
+    return yarnVersion;
+  } else {
+    if (caughtError.code === 'ENOENT') {
+      console.log(
+        'Could not find a Yarn installation. Please make sure Yarn is installed and ' +
+          'that the `yarn` command works in your command line shell.'
+      );
+    } else {
+      throw caughtError;
+    }
+
+    process.exit(1);
+  }
+}
+
+function gte(left, right) {
```

Comment:
This is not a correct implementation. For example `gte('2.0.0', '1.2.1')` says `false` but it should be `true`.

In general I'd prefer we don't reimplement this. Can we use something like `require('semver').gte` instead?